src/interface/src/app/treatments/metric-filters/metric-filters.component.ts

```diff
@@ -89,4 +108,78 @@ export class MetricFiltersComponent implements OnInit {
   activateMetric(metric: Metric, slot: MapMetricSlot): void {
     this.metricSelected.emit({ metric, slot });
   }
+
+  getTreatmentTypes() {
+    const options = [
+      { category: 'Single Treatment', options: [] },
+      { category: 'Sequenced Treatment', options: [] },
+    ];
+
+    this.treatmentState.summary$
+      .pipe(
+        filter((summary) => summary !== null),
+        take(1)
+      )
+      .subscribe((summary) => {
+        if (!summary?.project_areas) {
+          this.treatmentTypeOptions = options;
+          return;
+        }
+
+        summary.project_areas.forEach((project_area) => {
+          project_area.prescriptions.forEach((prescription) => {
+            this.addTreatmentOption(
+              prescription,
+              options,
+              PRESCRIPTIONS.SINGLE,
+              PRESCRIPTIONS.SEQUENCE
+            );
+          });
+        });
+
+        this.treatmentTypeOptions = options;
+      });
+  }
+
+  onConfirmedSelection(selection: any) {
+    this.directImpactsStateService.filteredTreatmentTypes$.next(
+      this.processTreatmentTypeSelection(selection)
+    );
+  }
+
+  private addTreatmentOption(
+    prescription: any,
+    options: { category: string; options: any[] }[],
+    singleActions: Record<string, string>,
+    sequencedActions: Record<string, SequenceAttributes>
+  ) {
+    if (singleActions[prescription.action]) {
+      options[0].options.push(singleActions[prescription.action]);
+    } else if (sequencedActions[prescription.action]) {
+      options[1].options.push(...sequencedActions[prescription.action].details);
+    }
+  }
+
+  processTreatmentTypeSelection(selection: string[]): string[] {
```

Comment:
One suggestion here, maybe if you type options of treatmentTypeOptions as an object or record with the PrescriptionActions and text, you might simplify this method? or even avoid it completely. Not sure if that something the sg-filter-dropdown supports.